File: backend/app/schemas/executive_dashboard.py
from datetime import date, datetime

from pydantic import BaseModel


class ExecutiveDashboardRankItem(BaseModel):
    label: str
    count: int


class ExecutiveDashboardEventItem(BaseModel):
    id: int
    title: str
    event_type: str
    severity: str
    status: str
    source_type: str | None = None
    source_id: str | None = None
    created_at: datetime


class ExecutiveDashboardPendingItem(BaseModel):
    id: int
    title: str
    priority: str
    status: str
    responsible_name: str | None = None
    due_date: date | None = None
    source_type: str | None = None
    source_id: str | None = None
    created_at: datetime


class ExecutiveDashboardSummary(BaseModel):
    total_events_today: int
    new_events: int
    high_events: int
    critical_events: int
    open_pending_items: int
    overdue_pending_items: int
    escalated_pending_items: int
    failed_routines_today: int
    top_projects_by_events: list[ExecutiveDashboardRankItem]
    top_responsibles_by_pending_items: list[ExecutiveDashboardRankItem]
    critical_event_list: list[ExecutiveDashboardEventItem]
    overdue_pending_item_list: list[ExecutiveDashboardPendingItem]

File: frontend/src/api/executiveDashboard.ts
import api from './client';

export interface ExecutiveRankItem {
  label: string;
  count: number;
}

export interface ExecutiveEventItem {
  id: number;
  title: string;
  event_type: string;
  severity: string;
  status: string;
  source_type?: string | null;
  source_id?: string | null;
  created_at: string;
}

export interface ExecutivePendingItem {
  id: number;
  title: string;
  priority: string;
  status: string;
  responsible_name?: string | null;
  due_date?: string | null;
  source_type?: string | null;
  source_id?: string | null;
  created_at: string;
}

export interface ExecutiveDashboardSummary {
  total_events_today: number;
  new_events: number;
  high_events: number;
  critical_events: number;
  open_pending_items: number;
  overdue_pending_items: number;
  escalated_pending_items: number;
  failed_routines_today: number;
  top_projects_by_events: ExecutiveRankItem[];
  top_responsibles_by_pending_items: ExecutiveRankItem[];
  critical_event_list: ExecutiveEventItem[];
  overdue_pending_item_list: ExecutivePendingItem[];
}

export const getExecutiveDashboardSummary = async () => {
  const { data } = await api.get<ExecutiveDashboardSummary>('/executive-dashboard/summary');
  return data;
};
